cmd/influx_inspect/export.go

```diff
@@ -136,70 +178,149 @@ func (c *cmdExport) writeFiles() error {
 
 	// Write out all the DDL
 	fmt.Fprintln(w, "# DDL")
-	for key := range c.files {
+	for key := range c.manifest {
 		keys := strings.Split(key, string(byte(os.PathSeparator)))
 		db, rp := influxql.QuoteIdent(keys[0]), influxql.QuoteIdent(keys[1])
 		fmt.Fprintf(w, "CREATE DATABASE %s WITH NAME %s\n", db, rp)
 	}
 
 	fmt.Fprintln(w, "# DML")
-	for key, files := range c.files {
+	for key := range c.manifest {
 		keys := strings.Split(key, string(byte(os.PathSeparator)))
+		fmt.Printf("writing out tsm file data for %s", key)
 		fmt.Fprintf(w, "# CONTEXT-DATABASE:%s\n", keys[0])
 		fmt.Fprintf(w, "# CONTEXT-RETENTION-POLICY:%s\n", keys[1])
-		for _, f := range files {
-			// use an anonymous function here to close the files in the defers and not let them
-			// accumulate in the loop
-			if err := func(f string) error {
-				file, err := os.OpenFile(f, os.O_RDONLY, 0600)
-				if err != nil {
-					return fmt.Errorf("%v", err)
-				}
-				defer file.Close()
-				reader, err := tsm1.NewTSMReader(file)
-				if err != nil {
-					log.Printf("unable to read %s, skipping\n", f)
-					return nil
+		if files, ok := c.tsmFiles[key]; ok {
+			fmt.Printf("writing out tsm file data for %s...", key)
+			c.writeTsmFiles(w, files)
+			fmt.Println("complete.")
+		}
+		if _, ok := c.walFiles[key]; ok {
+			fmt.Printf("writing out wal file data for %s...", key)
+			c.writeWALFiles(w, c.walFiles[key])
+			fmt.Println("complete.")
+		}
+	}
+	return nil
+}
+
+func (c *cmdExport) writeTsmFiles(w io.WriteCloser, files []string) error {
+	fmt.Fprintln(w, "# writing tsm data")
+	for _, f := range files {
+		// use an anonymous function here to close the files in the defers and not let them
+		// accumulate in the loop
+		if err := func(f string) error {
```

Comment:
This whole function could have its indentation significantly reduced by moving it out of the `for` loop. For example:

```go
do := func(fname string) error {
    // blah blah
}

for _, f := range files {
    if err := do(f); err != nil {
        return err
    }
}
return  nil
```